### ISSUE TITLE
Questions about paper and code

### ISSUE THREAD

**Issue by YicongHong** [ZhuFengdaaa/SOON#1]
Hi Fengda,

Thanks a lot for the interesting paper and the new dataset! I went through the code, got a few questions hopefully you can clarify for us -- to better follow your work! :smile: 

1. About Dataset
    1.1. From the paper, there are **Val Seen Instruction** and **Val Seen House** set, but the challenge website only provide **Val Unseen** data. Also, the paper does not report any **Val Unseen** results. So I am wondering if is it possible for you to release the **Val Seen** data, and the results of evaluating on the **Val Unseen Instruction** and **Val Unseen House** set. As reference for us to compare. (sorry I am very confused by the namings in paper Section 5.2 and Table 3 and the challenge website)
    1.2. Unlike REVERIE where the positions of objects are provided, so that people can extract regional features directly. The GBE method in your paper doesn't use regional features at all, but it direclty predict object heading and elevation from the ResNet-152 feature for the entire image. My concern is: since the object positions are also given in the data of SOON, we should assume that these positions are unknown to the network, right? Maybe you need to make this point clear for the challenge and the comparison in paper, otherwise, people might just do as in REVERIE -- directly use those positions to extract object features from the images, rather than running region proposal networks to predict bboxes.

2. About Training
    2.1. Arguments: Follow the paper, I believe the network is designed to predict heading and elevation of the target object directly, as default `args.compute` is False. However, `args.det_loss` should be set to True otherwise there is no loss for the localization, which does not appear in the given command `python3.6 -u r2r_src/train.py --upload --dataset SOON --maxAction 20 --log_every 500 --rl_ml_weight 1`.
    2.2. The paper mentioned using MSE loss to train localization, but the implementation (code) is in fact using BinaryCE loss, implying that the GT and prediction is normalized to (0,1) range. May I ask the reason for doing this normalization?
    2.3. The visual features should be ResNet-152 right? Not ResNet-101 as mentioned in your paper -- and in fact the visual encoder is not implemented -- since GBE uses Peter's preprocessed image features.
    
3. About Evaluation
    3.1. I would like to confirm my understanding here: The criteria of correctly finding an object is to check whether the predicted light ray representing as a point by `pre_point = Point(heading, elevation)` is in the virtual light rays polyon created by `gt_poly = Polygon(...)`. i.e. As long as the predicted point is inside the polygon, the localization is a success.

Thank you in advanced! Your anwser will be very helpful for us to follow your work!

**Comment by YicongHong** [comment]
Hi there,

Just to add something more about the data splits, I got the following results after running the complete training, using the provided arguments and data:

`BEST RESULT TILL NOW   val_unseen_instrs Iter 17000 `

`val_unseen_instrs , nav_error: 13.716, oracle_error: 7.696, steps: 13.612, lengths: 25.025, heading_errors: 0.193, elevation_errors: 0.032, point_det_errors: 0.145, goal_progress: 3.419, det_success_rate: 0.037, nav_success_rate: 0.230, oracle_rate: 0.296, spl: 0.200, success_rate: 0.037`

`val_unseen_house , nav_error: 11.791, oracle_error: 7.063, steps: 15.323, lengths: 30.028, heading_errors: 0.268, elevation_errors: 0.040, point_det_errors: 0.278, goal_progress: 5.369, det_success_rate: 0.006, nav_success_rate: 0.184, oracle_rate: 0.310, spl: 0.127, success_rate: 0.004`

I am not sure how to compare these results to the Table 3 in paper. 
(1) Does "Val_Unseen_Instrs" mean "seen house but unseen instructions"? (which it shouldn't because the numbers are way too smaller than the reported numbers in Table 3.)
(2) Does "Val_Unseen_House" mean "unseen house and unseen instructions"? (from Section 5.2, there is only one unseen validation set.)
(3) "Unseen House (Test)" on Table 3 is the test split reserved for the challenge right? Not the unseen validation set.

Thanks!

**Comment by ZhuFengdaaa** [comment]
Hi Yicong, 

Thanks a lot for your interest and helpful advice, 

1. About Dataset: The understanding of your second reply is correct: "Val_Unseen_Instrs" means "seen house but unseen instructions";  "Val_Unseen_House" means "unseen house and unseen instructions"; "Unseen House (Test)" on Table 3 is the test split reserved for the challenge, not the unseen validation set. The challenge results might be different from Table 3 since we labeled new data and re-split our dataset following the standard data split (we did not follow the standard train/val/test house split as VLN when we submit our paper)
2. About Training
    2.1 Arguments: Thanks for your correction, the command should be `python3.6 -u r2r_src/train.py --upload --dataset SOON --maxAction 20 --log_every 500 --rl_ml_weight 1 --det_loss True`. The README has been updated. 
    2.2 Intuitively we should use the MSE loss to learn. In practice, we find label normalization is helpful in training. This kind of implementation could also be referred to in https://arxiv.org/abs/1911.07883 (progress estimation task). Thanks for pointing out our mistake and we will update our arxiv paper to make it clear. 
    2.3 The visual features are ResNet-152, Peter's preprocessed image features. 
3. About Evaluation. Yes, your understanding is correct: as long as the predicted point is inside the polygon, the localization is a success. 

Thanks!

**Comment by YicongHong** [comment]
Thanks Fengda for the explanations, they are very helpful!

Hopefully I can come up with something fun for the challenge :laughing:
